src/bcrypt/__init__.py

```diff
@@ -76,3 +76,30 @@ def hashpw(password, salt):
         raise ValueError("Invalid salt")
 
     return _bcrypt.ffi.string(hashed)
+
+
+def kdf(password, salt, desired_key_bytes, rounds):
+    if isinstance(password, six.text_type) or isinstance(salt, six.text_type):
+        raise TypeError("Unicode-objects must be encoded before hashing")
+
+    if len(password) == 0 or len(salt) == 0:
+        raise ValueError("password and salt must not be empty")
+
+    if desired_key_bytes <= 0 or desired_key_bytes > 512:
```

Comment:
This feels like it can be a single equality + not, but I don't know if that ends up being more legible